src/colorizer/CanvasOverlay.ts

```diff
@@ -25,23 +25,28 @@ import {
 } from "./canvas/elements/annotations";
 import { BaseRenderParams, RenderInfo } from "./canvas/types";
 import { getPixelRatio } from "./canvas/utils";
+import { CanvasScaleInfo, CanvasType, FrameLoadResult } from "./types";
 
 import { LabelData } from "./AnnotationData";
-import Collection from "./Collection";
 import ColorizeCanvas from "./ColorizeCanvas";
-import ColorRamp from "./ColorRamp";
+import { IRenderCanvas, RenderCanvasStateParams } from "./IRenderCanvas";
 
 /**
- * Extends the ColorizeCanvas class by overlaying and compositing additional
- * dynamic elements (like a scale bar, timestamp, etc.) on top of the
+ * Wraps an IRenderCanvas class by overlaying and compositing additional
+ * dynamic elements (like a scale bar, timestamp, etc.) on top of a
  * base colorized image.
  */
-export default class CanvasWithOverlay extends ColorizeCanvas {
-  private canvas: HTMLCanvasElement;
+export default class CanvasOverlay implements IRenderCanvas {
+  private canvasElement: HTMLCanvasElement;
+  private canvas: IRenderCanvas;
   private ctx: CanvasRenderingContext2D;
 
-  private collection: Collection | null;
-  private datasetKey: string | null;
+  private currentFrame: number;
+  private params: RenderCanvasStateParams;
+  private onFrameLoadCallback: (result: FrameLoadResult) => void;
+
+  private zoomMultiplier: number;
+  private panOffset: Vector2;
```

Comment:
`CanvasOverlay` now tracks these values and passes them to the `canvas` when it's set.